src/coreclr/jit/assertionprop.cpp

```diff
@@ -3042,6 +3042,18 @@ GenTree* Compiler::optVNBasedFoldConstExpr(BasicBlock* block, GenTree* parent, G
             break;
         }
         break;
+
+        case TYP_MASK:
+        {
+            simdmask_t value = vnStore->ConstantValue<simdmask_t>(vnCns);
+
+            GenTreeVecCon* vecCon = gtNewVconNode(tree->TypeGet());
```

Comment:
Is `VecCon` really the right type to use here? This isn't a vector constant and I worry about other places potentially getting this wrong as they expect the type to only be `TYP_SIMD*`.

I'd have expected that we instead represent this is a new type, since its a completely separate kind of constant.

Reply:
Seems reasonable. I currently don't think we ever get here...